### ISSUE TITLE
Observe size on Map and Set

### ISSUE THREAD

**Issue by kamicane** [nx-js/observer-util#10]
Currently, it is not possible to observe the `size` property on Maps and Sets:

```js
const map = observable(new Map())

observe(() => console.log(map.size)) //0

setTimeout(() => map.set('key', 'value'), 200)

// crickets...
```

**Comment by solkimicreb** [comment]
Yes, thats true, I wil fix this in the next release (scheduled in a few days).

FYI: Maps and Sets are built-in objects, which need special treatment. Proxies can't be used with built-ins because they use internal (on JS) slots that can't be intercepted and that throws errors. Currently I patch them the old school way (by overwriting their methods). I will fix this be wrapping them on a Proxy which only intercepts when the key is not the name of a standard method. Otherwise it falls back to the old school patching.

TLDR: after the next update any property will be observable of Maps and Sets (including size)

Thx for the bug catch (:

**Comment by solkimicreb** [comment]
I fixed it with the next version (3.0.0). There are also a few breaking changes (the signal methods got moved to the library scope). Please read the readme for them. I also fixed some not yet raised issues (namely: expando props on es6 collections and $raw collections not being proper collections). Sorry for taking so long. The change log is coming later today.

Closing this now, thx for the issue (:
